protocols/gossipsub/src/behaviour.rs

```diff
@@ -771,7 +766,9 @@ impl Gossipsub {
                     }
                 }
             }
-            peers.retain(|peer| to_remove_peers.contains(&peer));
```

Comment:
This is different than in the old code, but I think the old code has the condition flipped.

Reply:
Yes true. Nice catch